connection.go

```diff
@@ -250,6 +251,49 @@ func (c *Connection) shutdown() {
 	close(c.errorChan)
 }
 
+// isConnectionReset checks if an error is a connection reset error
+func (c *Connection) isConnectionReset(err error) bool {
+	errStr := err.Error()
+	return strings.Contains(errStr, "connection reset") ||
+		strings.Contains(errStr, "broken pipe")
+}
+
+// checkProtocols checks if the protocols are explicitly stopped by the client- treat as normal connection closure
+func (c *Connection) checkProtocols() bool {
+	// Check chain-sync protocol
+	if c.chainSync != nil && (!c.chainSync.Client.IsDone() || !c.chainSync.Server.IsDone()) {
+		return false
+	}
+
+	// Check block-fetch protocol
+	if c.blockFetch != nil && (!c.blockFetch.Client.IsDone() || !c.blockFetch.Server.IsDone()) {
+		return false
+	}
+
+	// Check tx-submission protocol
+	if c.txSubmission != nil && (!c.txSubmission.Client.IsDone() || !c.txSubmission.Server.IsDone()) {
+		return false
+	}
```

Comment:
Let's also add `LocalStateQuery`, `LocalTxMonitor`, and `LocalTxSubmission` to this. It should be safe to do the NtN and NtC protocols all together with the `nil` checks.